src/axom/lumberjack/TextTagCombiner.hpp

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) 2017-2023, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+/*!
+ *******************************************************************************
+ * \file TextTagCombiner.hpp
+ *
+ * \brief This file contains the class implementation of the
+ * TextTagCombiner.
+ *******************************************************************************
+ */
+
+#ifndef TEXTTAGCOMBINER_HPP
+#define TEXTTAGCOMBINER_HPP
+
+#include "axom/lumberjack/Combiner.hpp"
+#include "axom/lumberjack/Message.hpp"
+
+#include <string>
+
+namespace axom
+{
+namespace lumberjack
+{
+/*!
+ *******************************************************************************
+ * \class TextTagCombiner
+ *
+ * \brief Combines Message classes if their Message::text and Message::tag
+ *        are equal.
+ *
+ *  This class can be added to Lumberjack's Lumberjack by calling
+ *  Lumberjack::addCombiner with a
+ *  TextTagCombiner instance as its parameter.
+ *
+ * \see Combiner Lumberjack
+ *******************************************************************************
+ */
+class TextTagCombiner : public Combiner
+{
+public:
+  TextTagCombiner() : m_id("TextTagCombiner") { }
+
+  /*!
+   *****************************************************************************
+   * \brief Returns the unique string identifier for this combiner. Used by
+   *  Lumberjack to differentiate between other combiners.
+   *****************************************************************************
+   */
+  const std::string id() { return m_id; }
```

Comment:
Since there's no way to change `m_id`, should it be a const static member variable?